### PR TITLE
Adding Drift V2 endpoints

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: proto/api.proto [hunk 1]
```diff
@@ -460,6 +474,17 @@ service Api {
     };
   }
 
+  rpc GetMarginOrderbook(GetMarginOrderbookRequest) returns (GetMarginOrderbookResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/market/Margin/orderbook/{contract}"
```
Comment (aspin) [inline proto/api.proto]: capital M => m

File: proto/api.proto [hunk 2]
```diff
@@ -378,6 +378,20 @@ service Api {
...
 
+  // Drift Margin
+  rpc PostMarginOrder(PostMarginOrderRequest)  returns (PostMarginOrderResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trade/margin/order"
+      body: "*",
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "post a margin order";
+      summary: "drift margin order";
```
Comment (aspin) [inline proto/api.proto]: remove `drift` from teh summary

File: proto/api.proto [hunk 3]
```diff
@@ -541,6 +566,17 @@ service Api {
...
   }
 
+  rpc GetMarginContracts(GetMarginContractsRequest) returns (GetMarginContractsResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/market/margin/contracts"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "Returns details of available perp contracts";
+      summary: "details of available perp contracts";
+      tags: ["Perp", "Market"];
```
Comment (aspin) [inline proto/api.proto]: remove `Perp` and add `Margin`

**Review by aspin — CHANGES_REQUESTED** [review]
some cosmetic labelling things

File: proto/api.proto (hunk 1 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

File: proto/api.proto (hunk 2 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

File: proto/api.proto (hunk 3 above)
Comment (tinystarinagalaxy) [reply proto/api.proto]: done

**Review by aspin — APPROVED** [review]

File: proto/api.proto [hunk 4]
```diff
@@ -74,6 +75,18 @@ service Api {
...
 
+  rpc GetMarketsV2(GetMarketsRequestV2) returns (GetMarketsResponseV2) {
+    option (google.api.http) = {
+      get: "/api/v2/spot/markets"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "Returns the list of orderbook markets";
+      summary: "Orderbook markets";
+      tags: ["Market", "Orderbook"];
```
Comment (aspin) [inline proto/api.proto]: we should tag all V1 endpoints as "v1" and all V2 endpoints as "v2"

Comment (aspin) [reply proto/api.proto]: we can also get rid of the "market" and "orderbook" tags, I think.

File: proto/api.proto [hunk 5]
```diff
@@ -51,6 +52,49 @@ service Api {
     }
   };
 
+  // Drift V2
```
Comment (aspin) [inline proto/api.proto:55]: consider moving everything v2 to the bottom of this service class. I think that'll be easier to read compared to having them interpersed

**Review by aspin — APPROVED** [review]

File: proto/api.proto (hunk 5 above)
Comment (tinystarinagalaxy) [reply proto/api.proto:55]: good idea, done

File: proto/api.proto [hunk 6]
```diff
@@ -724,6 +769,7 @@ message GetOrderbookRequest {
   string market = 1;
   uint32 limit = 2 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Use 0 for no limits"}];
   Project project = 3;
+  bool metadata = 4;
```
Comment (aspin) [inline proto/api.proto]: actually, I think this field is extraneous? since it's for Drift only and now that's on a different request

**Review by aspin — APPROVED** [review]

File: proto/api.proto [hunk 7]
```diff
@@ -51,6 +52,49 @@ service Api {
...
+  }
+
+  rpc GetDriftMarginOrderbook(GetDriftMarginOrderbookRequest) returns (GetDriftMarginOrderbookResponse) {
+    option (google.api.http) = {
+      get: "/api/v2/drift/margin-orderbooks/{market}"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      description: "Returns perp contract's orderbook. Use limit param to reduce the number of bids/asks returned";
+      summary: "Perpetuals contract orderbook";
+      tags: ["Perp", "Orderbook", "Market"];
```
Comment (aspin) [inline proto/api.proto:87]: missing `v2`, and should delete `perp`

File: proto/api.proto [hunk 8]
```diff
@@ -1678,3 +1744,58 @@ message GetOpenPerpOrderResponse {
...
+  string market = 1;
+  string marketAddress = 2;
+  repeated DriftMarginOrderbookItem bids = 3;
+  repeated DriftMarginOrderbookItem asks = 4;
+}
+
+message DriftMarginOrderbookItem {
+  double price = 1;
+  double size = 2;
+  string orderID = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Serum generated OrderID"}];
```
Comment (aspin) [inline proto/api.proto:1775]: reference to serum

**Pull request merged**